### DISCUSSION TITLE
Erro ao executar `docker-compose build`

### DISCUSSION THREAD

**Discussion by BrunoDivino** [rochacbruno/fastapi-workshop#1] [Q&A] [answered]
Olá pessoal!

Tenho tido problemas ao tentar executar o comando `docker-compose build`. Será que alguém pode me ajudar?

Estou no vídeo 03 - "Compose e Postgres" do workshop na LinuxTips.

Ao executar este comando, recebo uma série de erros. Resumidamente, são os seguintes:

```
ConnectionRefusedError: [Errno 111] Connection refused

urllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionRefusedError(111, 'Connection refused'))

requests.exceptions.ConnectionError: ('Connection aborted.', ConnectionRefusedError(111, 'Connection refused'))

docker.errors.DockerException: Error while fetching server API version: ('Connection aborted.', ConnectionRefusedError(111, 'Connection refused'))

```
Pelo que eu tenho pesquisado, isso pode ter a ver com as configurações das portas dos serviços e a conexão entre os containers. Porém, não sei ao certo como reparar a questão.

Imagem completa dos erros no terminal:

![image](https://user-images.githubusercontent.com/71059817/233877653-d023a263-a7f1-4924-84a9-89d8ffca0c9b.png)

Alguém tem alguma ideia?

**Comment by rochacbruno** [comment]
DEsconfio que pode ser algo na instalação do docker e permissões do seu utilizador ao docker.

pode tentar

```
sudo systemctl stop docker
sudo groupadd docker
sudo usermod -aG docker $USER
sudo su $USER
sudo systemctl start docker
docker-compose build

```

**Reply by BrunoDivino** [reply, marked as answer]
Olá Bruno! Agradeço a resposta.

Consegui resolver o problema hoje pela manhã.

Primeiramente, também suspeitei de uma possível instalação incorreta do docker. Para resolver isso, optei por uma solução simples e, alguns diriam, drástica. **Formatei o computador** para evitar conflitos de pacotes mal instalados. (Nada como uma bazuca para acabar com uma formiga.)

Porém, mesmo iniciando do zero e tendo instalado o Docker e docker-compose como recomendado na documentação, encontrei problemas ao utilizar o comando `docker-compose`. Contudo, ao tentar utilizar o comando `docker compose` (sem o traço) consegui realizar tudo que foi passado no workshop.

Ao pesquisar sobre isso, encontrei o seguinte trecho da documentação que sugere uma possível descontinuidade do comando `docker-compose`.


> Important

> From the end of June 2023 Compose V1 won’t be supported anymore and will be removed from all Docker Desktop versions.

> Make sure you switch to Compose V2 with the docker compose CLI plugin or by activating the Use Docker Compose V2 setting in Docker Desktop. For more information, see the Evolution of Compose.

> The new Compose V2, which supports the compose command as part of the Docker CLI, is now available.

> Compose V2 integrates compose functions into the Docker platform, continuing to support most of the previous docker-compose features and flags. You can run Compose V2 by replacing the hyphen (-) with a space, using docker compose, instead of docker-compose.


Fonte: https://docs.docker.com/compose/reference/

**A utilização de `docker compose` parece ter solucionado a questão de fato.** Obrigado pela atenção ;)

**Reply by rochacbruno** [reply]
Obrigado, vou adicionar essa nota a aula.
